### ISSUE TITLE
After objects are assigned, their fields don't have appropriate types

### ISSUE THREAD

**Issue by kelloggm** [kelloggm/checker-framework#127]
```java
import org.checkerframework.checker.index.qual.IndexFor;
import org.checkerframework.checker.index.qual.NonNegative;

public class ArrayAssignmentSameLenTest {

  static class Partial {
    private final int[] iValues;
    Partial(@NonNegative int n) {
      iValues = new int[n];
    }
  }

  private final Partial iBase;
  private final @IndexFor("iBase.iValues") int iFieldIndex;

  ArrayAssignmentSameLenTest(Partial partial, @IndexFor("#1.iValues") int fieldIndex) {
    iBase = partial;
    iFieldIndex = fieldIndex;
  }

}
```

First reported by @mernst as #124.

**Comment by kelloggm** [comment]
This problem affects all of our checkers, not just the SameLen Checker. Consider the following code:

```java
import org.checkerframework.checker.index.qual.*;

class ObjectAssignment {
    int x;
    
    static void do_things(ObjectAssignment a, ObjectAssignment b) {
	a.x = 5; // a.x is @Positive
	b = a; // b.x is now a.x, so it's also @Positive
	@Positive int y = b.x; // issues a warning
	@Positive int z = a.x; // does not issue a warning
    }
}
```

**Comment by kelloggm** [comment]
Seems to be a problem for checkers in general. If we handle it it should be at the framework level. Consider running the nullness checker on this code:

```java
import org.checkerframework.checker.nullness.qual.*;

class ObjectAssignment {

   @Nullable Object o;

   static void test_nullness(ObjectAssignment a, ObjectAssignment b) {
	a.o = new Object();
	b = a;
	int k = b.o.hashCode();
	int j = a.o.hashCode();
    }
}
```

**Comment by smillst** [comment]
Closed in favor of typetools#1149.
